content/en/blog/_posts/protect-mission-critical-pods-priorityclass/index.md

```diff
@@ -0,0 +1,283 @@
+        cpu: "2"
+```
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: prod-nginx
+  labels:
+    env: prod
+spec:
+  containers:
+  - name: prod-nginx
+    image: nginx
+    resources:
+      requests:
+        memory: "2Gi"
+        cpu: "2"
+      limits:
+        memory: "2Gi"
+        cpu: "2"
+```
+
+Let's create these pods by `kubectl create -f <FILE.yaml>` command and check their status by using `kubectl get pods`, if they are up and running to serve the traffic.
+
+```console
+$ kubectl get pods --show-labels
+NAME            READY   STATUS    RESTARTS   AGE   LABELS
+dev-nginx       1/1     Running   0          55s   env=dev
+preprod-nginx   1/1     Running   0          55s   env=preprod
+prod-nginx      0/1     Pending   0          55s   env=prod
+```
+
+We have bad news. Our prod pod is in pending status, which is bad since it won’t be serving any traffic, especially prod traffic.
+
+Let's see why this is happening:
+```console
+$ kubectl get events
+...
+...
+5s          Warning   FailedScheduling   pod/prod-nginx      0/2 nodes are available: 1 Insufficient cpu, 2 Insufficient memory.
+```
+
+To our surprise, our worker node is having some resource crunch.
+
+Now, let's look at how PriorityClass can help in this situation since prod should be given higher priority than the other environments.
+
+### PriorityClass resource
+
+Before we create the PriorityClasses based on our requirements, let's see what a basic YAML definition looks like and some prerequisites:
+
+```yaml
+apiVersion: scheduling.k8s.io/v1
+kind: PriorityClass
+metadata:
+  name: PRIORITYCLASS_NAME
+value: INT_VALUE_BETWEEN_0_TO_1000000000
+description: SMALL_DESCRIPTION
+globalDefault: TRUE_OR_FALSE
+```
+
+Below are some prerequisites for PriorityClasses:
+
+- The name of a PriorityClass must be a valid DNS subdomain name.
+- The name should not start with system-, the reason being that these names are already reserved by Kubernetes for its default PriorityClasses and to avoid unnecessary confusion.
+- Its value should be lower than or equal to 1 billion (1,000,000,000).
+- Larger numbers are reserved by PriorityClasses such as system-cluster-critical, system-node-critical for kube-system Pods.
+- There are two optional fields:
+  - _globalDefault:_ This value is used for pods where the priorityClassName is not specified and only one PriorityClass with globalDefault set to true can exist in a cluster.  
+  If there is no PriorityClass defined with globalDefault set to true, all the pods with no priorityClassName defined will be treated with 0 priority (i.e. the least priority).
+  - _Description:_ It's a string with a meaningful value so that people know when to use this PriorityClass.
+
+{{< note >}}
+Adding a PriorityClass with globalDefault set to true does not mean it will apply the same to the existing pods that are already running. This will be applicable only to the pods that came into existence after the PriorityClass was created.
+{{< /note >}}
```

Comment:
Honestly speaking since this was a critical point, I thought to call this out explicitly. If this doesn't look good please let me know will get this removed.